src/appengine/handlers/cron/fuzz_strategy_selection.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Fuzzing strategy selection cron job. 
+
+	Runs multi-armed bandit experiments for fuzzing strategy selection.
+	In particular, this is a Boltzman Exploration (softmax) implementaion
+  of multi-armed bandit experiments. Queries from bigquery to update
+  multi-armed bandit probability values based on the new edges for various
+  combined strategies. In the upload_bandit_weights function, we can change
+  metric to be for edges, crash, features, or units. Currently based on new edges."""
+
+from datastore import data_types
+from datastore import ndb
+from google_cloud_utils import big_query
+from handlers import base_handler
+from libs import handler
+
+
+
+
+"""BigQuery query for calculating multi-armed bandit probabilities for
+various strategies using a Boltzman Exploration (softmax) model.
+
+See https://www.cs.mcgill.ca/~vkules/bandits.pdf
+"""
+MULTI_ARMED_BANDIT_PROB = """
+SELECT
+  EXP(standardized_avg_ne / temp_p) / (SUM(EXP(standardized_avg_ne / temp_p)) OVER()) AS edges_bandit,
+  EXP(standardized_avg_cc / temp_p) / (SUM(EXP(standardized_avg_cc / temp_p)) OVER()) AS crash_bandit,
+  EXP(standardized_avg_nf / temp_p) / (SUM(EXP(standardized_avg_nf / temp_p)) OVER()) AS features_bandit,
+  EXP(standardized_avg_nu / temp_p) / (SUM(EXP(standardized_avg_nu / temp_p)) OVER()) AS units_bandit,
+  strategy,
+  strategy_count
+FROM (
+  SELECT
+    CONCAT(s_radamsa, s_max_len, s_ml_rnn, s_vp, s_corpus, s_fork, s_subset, s_recommended_dict) AS strategy,
+    AVG((crash_count - avg_cc) / stddev_cc) AS standardized_avg_cc,
+    AVG((new_edges - avg_ne) / stddev_ne) AS standardized_avg_ne,
+    AVG((new_features - avg_nf) / stddev_nf) AS standardized_avg_nf,
+    AVG((new_units_added - avg_nu) / stddev_nu) AS standardized_avg_nu,
+    COUNT(*) AS strategy_count,
+    "any" AS matcher,
+    /* Change temperature parameter here. */ .2 AS temp_p
```

Comment:
I don't think this comment will be necessary once the previous comment is addressed, but if you do want to keep it please move it to the end of the line.